File: UCF HSPT Documents/2010/Solutions/errata.cpp

#include <stdio.h>


int main(void)
{
   FILE *   fp;
   int      contestNum;
   int      numContests;
   int      numErrata;


   // Open the input file
   fp = fopen("errata.in", "r");

   // Read the number of contests
   fscanf(fp, "%d", &numContests);

   // Check each contest in the file
   for (contestNum = 1; contestNum <= numContests; contestNum++)
   {
      // Read the number of errata
      fscanf(fp, "%d", &numErrata);

      // Print the appropriate output
      if (numErrata == 0)
         printf("Contest #%d: No errata!\n", contestNum);
      else
         printf("Contest #%d: Not perfect, errata issued!\n", contestNum);
   }

   // Close the file before we go
   fclose(fp);

   // Return
   return 0;
}

File: UCF HSPT Team/src/y2010/errata.java

import java.io.*;
import java.util.Scanner;


public class errata
{
   public static void main(String args[]) throws Exception
   {
      Scanner   fp;
      int       contestNum;
      int       numContests;
      int       numErrata;

      // Open the input file
      fp = new Scanner(new File("errata.in"));

      // Read in the number of contests
      numContests = fp.nextInt();

      // Check each contest in the file
      for (contestNum = 1; contestNum <= numContests; contestNum++)
      {
         // Read in the number of errata in this contest
         numErrata = fp.nextInt();

         // Print the appropriate output
         if (numErrata == 0)
            System.out.println("Contest #" + contestNum + ": No errata!");
         else
            System.out.println(
               "Contest #" + contestNum + ": Not perfect, errata issued!");
      }
   }
}

